### PR TITLE
GUI: Plotly widget

### PR DESCRIPTION
**Pull request by tmichela**
follow up for #262

### CODE REVIEW

File: damnit/api.py [hunk 1]
```diff
@@ -87,6 +87,15 @@ def file(self) -> Path:
         """
         return self._h5_path
 
+    @property
+    def type_hint(self):
```
Comment (JamesWrigley) [inline damnit/api.py:91]: I'm inclined to make this a method because opening the file might be expensive.

File: damnit/gui/main_window.py [hunk 2]
```diff
@@ -983,6 +992,12 @@ def run_app(context_dir, connect_to_kafka=True):
             # User said no to setting up a new database
             return 0
 
+    # configure webviewer url engine
+    global SCHEME_HANDLER, WEB_PROFILE
```
Comment (JamesWrigley) [inline damnit/gui/main_window.py:996]: Does `SCHEME_HANDLER` need to be global? And can't we pass `WEB_PROFILE` to `PlotlyPlot()` directly?

**Review by JamesWrigley — CHANGES_REQUESTED** [review]

File: damnit/gui/main_window.py (hunk 2 above)
Comment (tmichela) [reply damnit/gui/main_window.py]: right, fixed

**Review by JamesWrigley — APPROVED** [review]
I'm a little nervous about depending on `PyQtWebEngine` (packaging can be a little horrendous), but c'est la vie. LGTM!

File: damnit/gui/web_viewer.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+from PyQt5 import QtCore, QtWidgets
```
Comment (CammilleCC) [inline damnit/gui/web_viewer.py:1]: Maybe clean up the imports? For instance, `QMovie` is not used.

File: damnit/gui/web_viewer.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+class PlotlyPlot(QtWidgets.QWidget):
+    def __init__(self, variable):
+        super().__init__()
+        self.setGeometry(0, 0, 1024, 768)
+
+        self.url = f"damn.it{variable._db.path.parent}/{variable.proposal}/{variable.run}/{variable.name}"
+
+        self.browser = QWebEngineView(self)
+        profile = QWebEngineProfile.defaultProfile()
```
Comment (CammilleCC) [inline damnit/gui/web_viewer.py:61]: Curious question: what's the difference of the profile that was declared on the `main_window` and the profile that was set here? Would it be possible to reuse it?

**Review by CammilleCC — COMMENTED** [review]
Looks good, haven't had the chance to test it locally but the logic seems sound. It's fascinating that one has to do the hacky way to implement a URI scheme just to show large HTML

File: damnit/gui/web_viewer.py (hunk 4 above)
Comment (tmichela) [reply damnit/gui/web_viewer.py:61]: they are the same. Here I get the [default profile](https://doc.qt.io/qt-6/qwebengineprofile.html#defaultProfile), I don't set a new one.

**Pull request merged**